src/storage/txn/process.rs

```diff
@@ -291,20 +298,24 @@ impl<E: Engine, S: MsgScheduler> Executor<E, S> {
                     }
                 } else {
                     let sched = scheduler.clone();
+                    let sched_pool = self.take_pool();
                     // The callback to receive async results of write prepare from the storage engine.
                     let engine_cb = Box::new(move |(_, result)| {
-                        notify_scheduler(
-                            sched,
-                            Msg::WriteFinished {
-                                cid,
-                                pr,
-                                result,
-                                tag,
-                            },
-                        );
-                        KV_COMMAND_KEYWRITE_HISTOGRAM_VEC
-                            .with_label_values(&[tag])
-                            .observe(rows as f64);
+                        sched_pool.pool.spawn(move || {
```

Comment:
em, do we still need this?

Reply:
how about enlarge the slot number from 1024 to 4096 to reduce lock contention?

Reply:
we still need this, because `notify_scheduler` can only be called in schedule worker threads (to get thread local engine). Later we can do some investigating about
1) why `Engine.clone` is so expensive
2) use future to avoid callbacks